src/services/highcharts-factory/highcharts-factory.svc.coffee

```diff
@@ -130,22 +118,48 @@ angular
       return unless thresholdSerie? && eventName? && _.isFunction(callback)
       Highcharts.addEvent(thresholdSerie, eventName, (_event)-> callback(thresholdSerie))
 
-    # Extend default chart formatters to add custom legend img icon
-    addCustomLegend: (formatterCallback, useHTML = true) ->
-      @hc.legend.update({
-        useHTML: useHTML
-        labelFormatter: formatterCallback
-      })
+    addCustomLegend: (labelFormatter, useHTML = true, showLegend = true) ->
+      legend =
+        legend:
+          labelFormatter: labelFormatter
+          useHTML: useHTML
+          enabled: showLegend
+      @options = angular.merge(@options, legend)
```

Comment:
The `angular.merge` method is destructive. The object in argument 1 will have the following object args merged into it. So @options will be mutated. If you want non destructive merging, pass an empty hash as the first argument like so:
```
# Note that @options will not contain attributes from someOtherHash
myNewHash = angular.merge({}, @options, someOtherHash)
```
So here you could simply write:
```
angular.merge(@options, legend)
```